CI isn't happy:
```
FAIL: geopmdpy_test/pytest_links/TestSecureFiles.test_creation_link_not_dir
======================================================================
FAIL: test_creation_link_not_dir (__main__.TestSecureFiles)
The path specified is a link not a directory.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/geopm/geopm/service/geopmdpy_test/TestSecureFiles.py", line 122, in test_creation_link_not_dir
    mock_sys_stderr_write.assert_has_calls(calls)
  File "/usr/lib/python3.10/unittest/mock.py", line 966, in assert_has_calls
    raise AssertionError(
AssertionError: Calls not found.
Expected: [call('Warning: <geopm-service> /tmp/tmpifadl4thTestSecureFiles/geopm-link is a symbolic link, the link will be renamed to /tmp/tmpifadl4thTestSecureFiles/geopm-link-uuid4-INVALID\n'),
 call('Warning: <geopm-service> the symbolic link points to /tmp\n')]
Actual: [call('Warning: <geopm-service> /tmp/tmpifadl4thTestSecureFiles/geopm-link is a symbolic link\n'),
 call('Warning: <geopm-service> the symbolic link points to /tmp\n'),
 call('Warning: <geopm-service> renamed invalid path /tmp/tmpifadl4thTestSecureFiles/geopm-link to /tmp/tmpifadl4thTestSecureFiles/geopm-link-uuid4-INVALID\n')]

FAIL: geopmdpy_test/pytest_links/TestSecureFiles.test_creation_file_not_dir
======================================================================
FAIL: test_creation_file_not_dir (__main__.TestSecureFiles)
The path specified is a file not a directory.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/geopm/geopm/service/geopmdpy_test/TestSecureFiles.py", line 147, in test_creation_file_not_dir
    mock_sys_stderr_write.assert_called_once_with(msg)
  File "/usr/lib/python3.10/unittest/mock.py", line 940, in assert_called_once_with
    raise AssertionError(msg)
AssertionError: Expected 'write' to be called once. Called 2 times.
Calls: [call('Warning: <geopm-service> /tmp/tmprgu60mqhTestSecureFiles/geopm.txt is not a directory\n'),
 call('Warning: <geopm-service> renamed invalid path /tmp/tmprgu60mqhTestSecureFiles/geopm.txt to /tmp/tmprgu60mqhTestSecureFiles/geopm.txt-uuid4-INVALID\n')].
```

--- a/service/geopmdpy/system_files.py
+++ b/service/geopmdpy/system_files.py
@@ -86,44 +86,43 @@ def _directory_permissions_are_correct(directory, perm_mode):
 
     set_perm_mode = stat.S_IMODE(st.st_mode)
     correct_permissions = set_perm_mode == perm_mode
-
-    user_owner = st.st_uid
-    correct_owner = user_owner == os.getuid()
-
-    group_owner = st.st_gid
-    correct_group = group_owner == os.getgid()
-
-    if correct_permissions and correct_owner and correct_group:
-        return True
     if not correct_permissions:
         sys.stderr.write(f'Warning: <geopm-service> {path} has wrong permissions, expected {oct(perm_mode)}\n')
         sys.stderr.write(f'Warning: <geopm-service> the wrong permissions were {oct(set_perm_mode)}\n')
+
+    user_owner = st.st_uid
+    correct_owner = user_owner == os.getuid()
     if not correct_owner:
         sys.stderr.write(f'Warning: <geopm-service> {path} has wrong user owner\n')
         sys.stderr.write(f'Warning: <geopm-service> the wrong user owner was {user_owner}\n')
+
+    group_owner = st.st_gid
+    correct_group = group_owner == os.getgid()
     if not correct_group:
         sys.stderr.write(f'Warning: <geopm-service> {path} has wrong group owner\n')
         sys.stderr.write(f'Warning: <geopm-service> the wrong group owner was {group_owner}\n')
-    return False
+
+    return correct_permissions and correct_owner and correct_group
 
 
 def _is_already_secure_directory(candidate_dir, perm_mode):
     """Helper function to check if a path is a directory with the correct permissions and ownership
     """
     path = candidate_dir
+    is_secure_directory = False
     # If it's a link
     if os.path.islink(path):
         sys.stderr.write(f'Warning: <geopm-service> {path} is a symbolic link\n')
         sys.stderr.write(f'Warning: <geopm-service> the symbolic link points to {os.readlink(path)}\n')
-        return False
     # If it's a directory
-    if os.path.isdir(path):
+    elif os.path.isdir(path):
         if _directory_permissions_are_correct(path, perm_mode):
-            return True
+          is_secure_directory = True
+    # Don't know what it is, definitely not a directory
     else:
         sys.stderr.write(f'Warning: <geopm-service> {path} is not a directory\n')
     # default condition: path is not a secure directory
-    return False
+    return is_secure_directory
 
 
 def secure_make_dirs(path, perm_mode=0o700):

--- a/service/geopmdpy_test/TestSecureFiles.py
+++ b/service/geopmdpy_test/TestSecureFiles.py
@@ -116,8 +116,9 @@ class TestSecureFiles(unittest.TestCase):
             secure_make_dirs(sess_path)
             renamed_path = f'{sess_path}-uuid4-INVALID'
             calls = [
-                mock.call(f'Warning: <geopm-service> {sess_path} is a symbolic link, the link will be renamed to {renamed_path}\n'),
-                mock.call(f'Warning: <geopm-service> the symbolic link points to /tmp\n')
+                mock.call(f'Warning: <geopm-service> {sess_path} is a symbolic link\n'),
+                mock.call(f'Warning: <geopm-service> the symbolic link points to /tmp\n'),
+                mock.call(f'Warning: <geopm-service> renamed invalid path {sess_path} to {renamed_path}\n')
             ]
             mock_sys_stderr_write.assert_has_calls(calls)
             mock_os_path_islink.assert_called_once_with(sess_path)
@@ -143,8 +144,11 @@ class TestSecureFiles(unittest.TestCase):
              mock.patch('sys.stderr.write', return_value=None) as mock_sys_stderr_write:
             secure_make_dirs(sess_path)
             renamed_path = f'{sess_path}-uuid4-INVALID'
-            msg = f'Warning: <geopm-service> {sess_path} is not a directory, it will be renamed to {renamed_path}\n'
-            mock_sys_stderr_write.assert_called_once_with(msg)
+            calls = [
+                mock.call(f'Warning: <geopm-service> {sess_path} is not a directory\n'),
+                mock.call(f'Warning: <geopm-service> renamed invalid path {sess_path} to {renamed_path}\n')
+            ]
+            mock_sys_stderr_write.assert_has_calls(calls)
             mock_os_path_islink.assert_called_once_with(sess_path)
             mock_os_path_isdir.assert_called_once_with(sess_path)
         self.assertTrue(os.path.exists(renamed_path))

